### PR TITLE
kitty: 0.32.2 -> 0.33.0

### PR DESCRIPTION
**Pull request by leiserfg**
## Description of changes
https://sw.kovidgoyal.net/kitty/changelog/#detailed-list-of-changes

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by leiserfg** [comment]
I had to bump the go used to build the kittens because of changes upstream.

File: pkgs/applications/terminal-emulators/kitty/default.nix [hunk 1]
```diff
@@ -18,37 +18,39 @@
 , libpng
 , python3
 , zlib
+, simde
 , bashInteractive
 , zsh
 , fish
 , nixosTests
-, go
+, go_1_22
```
Comment (adamcstephens) [inline pkgs/applications/terminal-emulators/kitty/default.nix:26]: Why change from the default go? I don't see anything about this in the release notes.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -3030,6 +3030,7 @@ with pkgs;
       UniformTypeIdentifiers
       UserNotifications
     ;
+    buildGoModule = buildGo122Module;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:3033]: Maybe we can move this into the file to easier keep it in sync with the go used there

**Comment by adamcstephens** [comment]
Note that the Darwin build is failing

**Comment by leiserfg** [comment]
> Note that the Darwin build is failing

How is it? I don't have a mac to test.

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (leiserfg) [reply pkgs/top-level/all-packages.nix]: I just did as in other packages but yes I also prefer to move it.

**Comment by adamcstephens** [comment]
> How is it? I don't have a mac to test.

see the ofborg output in the checks.

**Comment by leiserfg** [comment]
> see the ofborg output in the checks.

error: a 'aarch64-linux' with features {} is required to build '/nix/store/70vb73asfydfxff8c7ksw2ny1d4gpfvj-Xresources-Xft.drv', but I am a 'aarch64-darwin' with features {benchmark, big-parallel, nixos-test}

Looks more like an ofborg issue to me (zero experience with it to be honest).

**Comment by adamcstephens** [comment]
No, something has changed in the tests. Can you please update your commit with:

```diff
   passthru = {
-    tests.test = nixosTests.terminal-emulators.kitty;
+    tests = lib.mkIf stdenv.isLinux {
+      default = nixosTests.terminal-emulators.kitty;
+    };
     updateScript = nix-update-script {};
   };
```

**Comment by leiserfg** [comment]
Should I make a new PR for it or will you fix it?

**Pull request merged**